### PR TITLE
[#50] Migrate to ES6

### PR DESCRIPTION
**Pull request by vasco3**

### CODE REVIEW

File: app/node_modules/server/index.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-var
+import express from 'express';
+import compress from 'compression';
+import logger from 'bunyan-request-logger';
+import noCache from 'connect-cache-control';
+
+const app = express(),
   log = logger(),
-  pubPath = __dirname + '/../public';
+	pubPath = __dirname + '/../public';
```
Comment (ericelliott) [inline app/node_modules/server/index.js]: There should be two spaces at the beginning of this line.

File: app/test/unit/healthcheck/index.js [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
+const
+  pkg = require(`${root}/package.json`),
+  buildPath = `${root}/config/BUILD`,
   build = fs.readFileSync(buildPath, 'utf8').trim();
 
-module.exports = function client () {
+export default function client () {
 
-  test('Healthcheck server', function (assert) {
+  test('Healthcheck server', (assert) => {
```
Comment (ericelliott) [inline app/test/unit/healthcheck/index.js]: You don't need parens around `assert` here.

**Comment by ericelliott** [comment]
Make sure `npm run check` works.

File: app/node_modules/server/index.js (hunk 1 above)
Comment (nkbt) [inline app/node_modules/server/index.js]: I would also avoid stacking declarations and use one declaration per line.

File: app/test/unit/healthcheck/index.js [hunk 3]
```diff
@@ -1,27 +1,26 @@
 'use strict';
```
Comment (nkbt) [inline app/test/unit/healthcheck/index.js]: Remove `use strict`. It is automatically applied in case of ES6 modules

**Comment by vasco3** [comment]
I'm getting the following error in bunyan after migrating to es6. I've seen that before, and it is fixed in the new bunyan version. So I submitted a PR for bunyan-request-logger https://github.com/ericelliott/bunyan-request-logger/pull/13

The error is 
```

> universal-react-boilerplate@1.2.3 test /var/repositories/universal-react-boilerplate
> babel-node --ignore '/node_modules/(?!server|client|lib|healthcheck)' app/test/unit/index.js | tap-bail

/var/repositories/universal-react-boilerplate/node_modules/bunyan-request-logger/node_modules/bunyan/lib/bunyan.js:1348
module.exports.RotatingFileStream = RotatingFileStream;
                                    ^
ReferenceError: RotatingFileStream is not defined
    at Object.<anonymous> (/var/repositories/universal-react-boilerplate/node_modules/bunyan-request-logger/node_modules/bunyan/lib/bunyan.js:1429:37)
    at Module._compile (module.js:460:26)
    at normalLoader (/var/repositories/universal-react-boilerplate/node_modules/babel-core/lib/api/register/node.js:199:5)
    at Object.require.extensions.(anonymous function) [as .js] (/var/repositories/universal-react-boilerplate/node_modules/babel-core/lib/api/register/node.js:216:7)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/var/repositories/universal-react-boilerplate/node_modules/bunyan-request-logger/request-logger.js:5:12)
    at Module._compile (module.js:460:26)
```

**Comment by vasco3** [comment]
Bunyan got fixed. But now babel-node is complaining about node-cuid.

```
> babel-node --ignore '/node_modules/(?!server|client|lib|healthcheck)' app/test/unit/index.js | tap-bail

/var/repositories/universal-react-boilerplate/node_modules/bunyan-request-logger/node_modules/cuid/dist/node-cuid.js:92
})(undefined.applitude || undefined);
            ^
TypeError: Cannot read property 'applitude' of undefined
    at Object.<anonymous> (/var/repositories/universal-react-boilerplate/node_modules/bunyan-request-logger/node_modules/cuid/dist/node-cuid.js:105:3)
    at Module._compile (module.js:460:26)
    at normalLoader (/var/repositories/universal-react-boilerplate/node_modules/babel-core/lib/api/register/node.js:199:5)
    at Object.require.extensions.(anonymous function) [as .js] (/var/repositories/universal-react-boilerplate/node_modules/babel-core/lib/api/register/node.js:216:7)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/var/repositories/universal-react-boilerplate/node_modules/bunyan-request-logger/request-logger.js:8:10)
    at Module._compile (module.js:460:26)
```

**Comment by ericelliott** [comment]
`cuid` is in a weird state right now. @therealklanni took a crack at modernizing it, but that broke it, so we rolled back, and I haven't had time to get it fixed. Would you like to take a look at that, too?

=)

**Pull request closed without merging**